backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLineBL.java

```diff
@@ -1033,4 +1044,58 @@ public Optional<BPartnerId> getBPartnerId(@NonNull final OrderAndLineId orderLin
 		return BPartnerId.optionalOfRepoId(orderLine.getC_BPartner_ID());
 	}
 
+	@Override
+	public void setTax(@NonNull final org.compiere.model.I_C_OrderLine  orderLine)
+	{
+		final I_C_Order orderRecord = orderBL().getById(OrderId.ofRepoId(orderLine.getC_Order_ID()));
+
+		final TaxCategoryId taxCategoryId = getTaxCategoryId(orderLine);
+
+		final WarehouseId warehouseId = warehouseAdvisor.evaluateWarehouse(orderLine);
+		final CountryId countryFromId = warehouseBL.getCountryId(warehouseId);
+
+		final BPartnerLocationAndCaptureId bpLocationId = OrderLineDocumentLocationAdapterFactory.locationAdapter(orderLine).getBPartnerLocationAndCaptureId();
+
+		final boolean isSOTrx = orderRecord.isSOTrx();
+		final Timestamp taxDate = orderLine.getDatePromised();
+
+		final BPartnerId effectiveBillPartnerId = orderBL().getEffectiveBillPartnerId(orderRecord);
+		final Boolean isTaxExempt;
+		if (effectiveBillPartnerId != null)
+		{
+			final I_C_BPartner billBPartnerRecord = bpartnerDAO.getById(effectiveBillPartnerId);
+			isTaxExempt = billBPartnerRecord.isTaxExempt();
```

Comment:
We can't **dissallow** tax-exempt `C_Taxes` in case the `C_B_Partner` has the flag not set.
Y => filter for tax-exempt tax
N => like null, do not filter